rs/sns/governance/src/governance.rs

```diff
@@ -5541,93 +5562,122 @@ impl Governance {
                 .clone_from(&target_version.archive_wasm_hash);
         }
 
-        let (current_version, new_cached_upgrade_steps) = match self
-            .proto
-            .cached_upgrade_steps_or_err()
-            .and_then(CachedUpgradeSteps::consume)
-        {
-            Err(err) => {
-                let error = format!(
-                    "Keeping the currently running version and marking the upgrade as failed \
-                     at timestamp {} seconds: {}",
+        // TODO: DO NOT MERGE
+        //
+        // This condition is simplified, but it does not take into account the fact that Ledgers
+        // may be of a different version than Archives. This is due to the fact that a Ledger
+        // is responsible for spawning archives using the sources stored in its own memory, whereas
+        // currently the Ledger is *not* responsible for upgrading the archives it had previously
+        // spawned. Due to this unfortunate circumstances, the SNS needs to take special care to
+        // treat the case of upgrading the archives.
+        //
+        // If in doubt, revert the change in this condition to the code on the master branch.
+        if running_version == target_version {
+            self.set_proposal_execution_status(proposal_id, Ok(()));
+
+            self.proto.pending_version = None;
+
+            self.push_to_upgrade_journal(upgrade_journal_entry::UpgradeOutcome {
+                status: Some(upgrade_journal_entry::upgrade_outcome::Status::Success(
+                    Empty {},
+                )),
+                human_readable: Some(format!(
+                    "Marking upgrade SUCCESS at {} seconds.",
+                    self.env.now()
+                )),
+            });
```

Comment:
When you rebase / before you merge into master, could you change this to use the helpers that are in `src/upgrade_journal.rs`? And ditto elsewhere you call `push_to_upgrade_journal`.